pyat/at/physics/nonlinear.py

```diff
@@ -193,12 +235,165 @@ def gen_detuning_elem(ring: Lattice, orbit: Optional[Orbit] = None) -> Element:
         orbit, _ = find_orbit(ring)
     lindata0, _, _ = ring.linopt6(get_chrom=False, orbit=orbit)
     xsi = get_chrom(ring.radiation_off(copy=True))
-    r0, r1, x, q_dx, y, q_dy = detuning(ring.radiation_off(copy=True),
-                                        xm=1.0e-4, ym=1.0e-4, npoints=3)
-    nonlin_elem = Element('NonLinear', PassMethod='DeltaQPass',
-                          Betax=lindata0.beta[0], Betay=lindata0.beta[1],
-                          Alphax=lindata0.alpha[0], Alphay=lindata0.alpha[1],
-                          Qpx=xsi[0], Qpy=xsi[1],
-                          A1=r1[0][0], A2=r1[0][1], A3=r1[1][1],
-                          T1=-orbit, T2=orbit)
+    r0, r1, x, q_dx, y, q_dy = detuning(
+        ring.radiation_off(copy=True), xm=1.0e-4, ym=1.0e-4, npoints=3
+    )
+    nonlin_elem = Element(
+        "NonLinear",
+        PassMethod="DeltaQPass",
+        Betax=lindata0.beta[0],
+        Betay=lindata0.beta[1],
+        Alphax=lindata0.alpha[0],
+        Alphay=lindata0.alpha[1],
+        Qpx=xsi[0],
+        Qpy=xsi[1],
+        A1=r1[0][0],
+        A2=r1[0][1],
+        A3=r1[1][1],
+        T1=-orbit,
+        T2=orbit,
+    )
     return nonlin_elem
+
+
+def feeddown_polynomba(
+    polb: numpy.ndarray or tuple = (),
```

Comment:
Don;t we want to allow a list as well? Maybe the type should be Sequence[float]

Reply:
Changed to Sequence[float]